dispatcher/dispatcherV2.go

```diff
@@ -0,0 +1,250 @@
+package dispatcher
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	goproto "github.com/iotexproject/iotex-proto/golang"
+	"github.com/iotexproject/iotex-proto/golang/iotexrpc"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/iotexproject/iotex-core/pkg/lifecycle"
+	"github.com/iotexproject/iotex-core/pkg/log"
+)
+
+type (
+	dispatcherV2 struct {
```

Comment:
adding `msgQueueMgr` is a good move, to handle the message dispatch into different queue 
on top of that, i think you can directly modify current dispatcher, no need to add a V2, so we can clearly see the changes

Reply:
combined the v2 with v1